lib/mongoid/fields.rb

```diff
@@ -561,6 +562,7 @@ def add_defaults(field)
       #
       # @api private
       def add_field(name, options = {})
+        @database_field_names[name] = nil if @database_field_names
```

Comment:
I'm curious what the purpose of this line is; is this resetting some previously set field? If it's just for initialization, I'd prefer to leave it out, since the field would be nil implicitly, by default.

Reply:
It's meant as some sort of cache busting.
This is in case something unexpected like adding a field happens after we have accessed that field, for example, we have accessed field `abc` earlier as an alias, but later in the lifecycle, somehow that field added as an actual field.
I'm not certain about the exact flow, but perhaps we should also revoke the value for any alias that field might have.